fe/fe-core/src/main/java/org/apache/doris/nereids/rules/implementation/LogicalPartitionTopNToPhysicalPartitionTopN.java

```diff
@@ -21,33 +21,68 @@
 import org.apache.doris.nereids.rules.Rule;
 import org.apache.doris.nereids.rules.RuleType;
 import org.apache.doris.nereids.trees.expressions.OrderExpression;
+import org.apache.doris.nereids.trees.plans.PartitionTopnPhase;
+import org.apache.doris.nereids.trees.plans.Plan;
+import org.apache.doris.nereids.trees.plans.logical.LogicalPartitionTopN;
 import org.apache.doris.nereids.trees.plans.physical.PhysicalPartitionTopN;
 
-import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
 
 import java.util.List;
+import java.util.stream.Collectors;
 
 /**
  * Implementation rule that convert logical partition-top-n to physical partition-top-n.
  */
 public class LogicalPartitionTopNToPhysicalPartitionTopN extends OneImplementationRuleFactory {
     @Override
     public Rule build() {
-        return logicalPartitionTopN().then(partitionTopN -> {
-            List<OrderKey> orderKeys = !partitionTopN.getOrderKeys().isEmpty()
-                    ? partitionTopN.getOrderKeys().stream()
-                        .map(OrderExpression::getOrderKey)
-                        .collect(ImmutableList.toImmutableList()) :
-                    ImmutableList.of();
-
-            return new PhysicalPartitionTopN<>(
-                    partitionTopN.getFunction(),
-                    partitionTopN.getPartitionKeys(),
-                    orderKeys,
-                    partitionTopN.hasGlobalLimit(),
-                    partitionTopN.getPartitionLimit(),
-                    partitionTopN.getLogicalProperties(),
-                    partitionTopN.child());
-        }).toRule(RuleType.LOGICAL_PARTITION_TOP_N_TO_PHYSICAL_PARTITION_TOP_N_RULE);
+        return logicalPartitionTopN().thenApplyMulti(ctx -> twoPhasePartitionTopn(ctx.root))
```

Comment:
if `twoPhasePartitionTopn` only return one candidate, just use `thenApply` instead of `thenApplyMulti`